### PR TITLE
Refactor Controllers out of the SCENE

### PR DESCRIPTION
**Pull request by ttyz**
This is a proof of concept for #403 and #404. SoundController is no longer a MonoBehaviour. Which allows to remove it from the SCENE (I did not do it yet though).

### CODE REVIEW

**Comment by ttyz** [comment]
All the easy ones are done :) Mouse, FurnitureSprite and BuildMode are a bit more tricky, so will do them later. If someone finds this useful in it's current state, it may be merged. Also would like to move everything to a `GameController` from the World, but also might be a separate PR.

**Comment by dustinwloring1988** [comment]
@ttyz - I would like to see this get fixed and added to the project I tested it a couple commits ago and I really liked the clean up of controllers.

**Comment by ttyz** [comment]
Fixed.

File: Assets/Scripts/Controllers/MouseController.cs [hunk 1]
```diff
@@ -417,7 +421,7 @@ void ShowFurnitureSpriteAtTile(string furnitureType, Tile t)
     {
 
         GameObject go = new GameObject();
-        go.transform.SetParent(this.transform, true);
+//FIXME        go.transform.SetParent(this.transform, true);
```
Comment (alkemann) [inline Assets/Scripts/Controllers/MouseController.cs]: Should this not be fixed before PR merge?

Comment (ttyz) [inline Assets/Scripts/Controllers/MouseController.cs]: Yes, you are 100% correct. Will fix ASAP

**Comment by dustinwloring1988** [comment]
Nice cant wait until it gets added I wish I could accept Pulls

**Comment by ttyz** [comment]
I will be less active here during the week. But I really want to get this merged, please notify me if this gets out of sync or there are any issues with style or bugs, I will try to correct it.

**Comment by vogonistic** [comment]
@ttyz It has a merge conflict now

**Comment by ttyz** [comment]
Cleaned those up. Ready for merge

**Comment by vogonistic** [comment]
I'm getting a lot of warnings and errors when trying to test this. I'm also seeing all the objects in the scene. Can you double-check your fix?

**Comment by ttyz** [comment]
I always forget that Unity does not auto save the SCENE

**Comment by vogonistic** [comment]
Merged!

**Pull request merged**